plugin-api/src/test/java/org/sonar/api/server/rule/internal/DefaultRuleTest.java

```diff
@@ -92,6 +96,78 @@ public void getters() {
     assertThat(defaultRule.educationPrincipleKeys()).containsOnly("principle_key1", "principle_key2", "principle_key3");
   }
 
+  @Test
+  public void constructor_impact_is_set_when_type_and_severity_is_null() {
+    DefaultNewRule rule = new DefaultNewRule("plugin", "repo", "key");
+
+    DefaultRule defaultRule = new DefaultRule(repo, rule);
+
+    assertThat(defaultRule.defaultImpacts()).containsEntry(SoftwareQuality.MAINTAINABILITY, Severity.MEDIUM);
```

Comment:
We might want to assert type and severity also

Reply:
done